### DISCUSSION TITLE
Issues about SideDiffusiveFluxIntegral postprocessor

### DISCUSSION THREAD

**Discussion by WayneX-Wong** [idaholab/moose#23052] [Q&A General] [answered]
Hi all,

I ran a simple steady-state heat conduction problem with a circular cylindrical shell model, heat source set to 21.057 W/cm3 (volume of the model is 47.5 cm3, so total power is 1000 W).

The top, bottom and outer surfaces of the model were set as adiabatic boundary conditions, and the inner surfaces were set as temperature Dirichlet boundary.

I added SideDiffusiveFluxIntegral postprocessor on both inner and outer boundary, and I think theoretically the former should be equal to the total power, e.g. 1000, but it gives 867 (and will vary slightly with given boundary temperature), and the latter gives -122.

Does anyone have a clue about this?

Here is my input
```
[Mesh]
  [file]
    type = FileMeshGenerator
    file = simp.e
  []
[]

[Variables]
  [temp]
  []
[]

[Kernels]
  [hc]
    type = HeatConduction
    variable = temp
  []
  [heat]
    type = HeatSource
    variable = temp
    value = 21.052
  []
[]

[BCs]
  [heat_pipe_wall_temp]
    type = DirichletBC
    boundary = 'inner'
    variable = temp
    value = 900
  []
  [adiabatic_surface]
    type = NeumannBC
    boundary = 'top bottom outer'
    variable = temp
  []
[]

[Materials]
  [k_fuel]
    type = ParsedMaterial
    args = 'temp'
    block = 'heater'
    f_name = thermal_conductivity
    function = '(12.17+0.038*(temp-273.15))/100'
  []
[]

[Executioner]
  type = Transient
  nl_abs_tol = 1e-8
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
  steady_state_detection = true
  steady_state_tolerance = 1e-08
[]

[Postprocessors]
  [flux_integral_inner]
    type = SideDiffusiveFluxIntegral
    boundary = 'inner'
    variable = temp
    diffusivity = 'thermal_conductivity'
  []
  [flux_integral_outer]
    type = SideDiffusiveFluxIntegral
    boundary = 'outer'
    variable = temp
    diffusivity = 'thermal_conductivity'
  []
  [max_T]
    type = NodalExtremeValue
    variable = temp
  []
[]

[Outputs]
  exodus = true
  print_linear_residuals = false
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

It seems odd to have a non zero flux with a Neumann BC of 0.
Is the result converged? Can you please attach the output log
Does this get better with mesh refinement?

Guillaume

**Reply by WayneX-Wong** [reply]
Ah...I didn't realize it was the mesh which lead to this problem before.

I refine the mesh and the result seems better.

Thanks for your guidance Guillaume,  Sorry for taking up your time for this minor problem

**Reply by GiudGiud** [reply]
no worries
